pkg/api/rest/config.go

```diff
@@ -0,0 +1,77 @@
+package rest
+
+import (
+	"fmt"
+	"net/url"
+	"path"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/ticdc/pkg/httputil"
+	"github.com/pingcap/ticdc/pkg/security"
+	"golang.org/x/time/rate"
+)
+
+// Config holds the common attributes that can be passed to a cdc REST client
+type Config struct {
+	// Host must be a host string, a host:port pair, or a URL to the base of the cdc server.
+	Host string
+	// APIPath is a sub-path that points to an API root.
+	APIPath string
+	// Credential holds the security Credential used for generating tls config
+	Credential *security.Credential
+	// Rate limiter for limiting connections to the cdc owner from this client.
+	RateLimiter *rate.Limiter
+	// Api verion
+	Version string
+}
+
+// defaultServerUrlFor is used to build base URL and api path.
+func defaultServerUrlFor(config *Config) (*url.URL, string, error) {
+	host := config.Host
+	if host == "" {
+		host = "127.0.0.1"
+	}
+	base := host
+	hostURL, err := url.Parse(base)
+	if err != nil || hostURL.Scheme == "" || hostURL.Host == "" {
+		scheme := "http://"
+		if config.Credential != nil && config.Credential.IsTLSEnabled() {
+			scheme = "https://"
+		}
+		hostURL, err = url.Parse(scheme + base)
+		if err != nil {
+			return nil, "", errors.Trace(err)
+		}
+		if hostURL.Path != "" && hostURL.Path != "/" {
+			return nil, "", fmt.Errorf("host must be a URL or a host:port pair: %q", base)
+		}
+	}
+	versionedPath := path.Join("/", config.APIPath, config.Version)
+	return hostURL, versionedPath, nil
+}
+
+func RESTClientForConfig(config *Config, httpClient *httputil.Client) (*RESTClient, error) {
```

Comment:
Can `RESTClientForConfig` and `RESTClientFor` merge into one func?

Reply:
RESTClientForConfig accepts a `httpClient` parameter which is useful for customizing a fake http client. I will provide a fake REST client in Part Ⅲ.

Reply:
```suggestion
func RESTClientFromHttpClient(config *Config, httpClient *httputil.Client) (*RESTClient, error) {
```
Also can you explain the motivation for this function in comments?

Reply:
It will be used in PR part 2: https://github.com/pingcap/tiflow/blob/27643133813709878ba260166f7a959a65564f63/pkg/api/v1/api_client.go#L42-L55

Reply:
Previously, i assume the httpClient here can be customized by users. But at present, the `RESTClientFor` and `RESTClientForConfig` functions can be merged into one. I will modify it later.